src/test/ui/async-await/issue-61949-self-return-type.stderr

```diff
@@ -0,0 +1,8 @@
+error: `async fn` return type cannot contain a projection or `Self` that references lifetimes from a parent scope
```

Comment:
I wonder if we should try to give some clue for how to workaround this -- this is *basically* a rustc bug. 

In the case of `Self`, the suggestion would be to write the `Self` type explicitly.

Reply:
I've dug around for a bit and I've not been able to find a way to determine if the return type was originally `Self` from here. I'm happy to add a suggestion if anyone knows a way to determine this.

Reply:
At minimum we could just -- for this particular issue -- give it an error code and direct them to some chapter in the async book where we explain in more depth (in the extended explanation). But we'd need a link first. Maybe we can just file a follow-up issue on the matter to improve diagnostics.